draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
-
-The property `sd_digests` MUST be used by the Issuer to include digests of the salted claim values for any claim that is intended to be selectively disclosable.
+### Selectively Disclosable Claims
 
-The Issuer MUST choose a new, cryptographically random salt value for
-each claim value. The salt value MUST then be encoded as a string. It is
-RECOMMENDED to base64url-encode the salt value.
+For each claim that is to be selectively disclosed, the Issuer creates a Disclosure object, hashes it, and includes the hash instead of the original claim in the SD-JWT, as described next. The Disclosures are then sent to the Holder.
 
-The Issuer MUST generate the digests over a JSON literal according to
-[@!RFC8259] that is formed by
-JSON-encoding an object with the following contents:
+#### Creating Disclosures {#creating_disclosures}
+The Issuer MUST create a Disclosure object for each selectively disclosable claim as follows:
 
- * REQUIRED with the key `s`: the salt value,
- * REQUIRED with the key `v`: the claim value (either a string or a more complex object, e.g., for the [@OIDC] `address` claim),
- * OPTIONAL, with the key `n`: the claim name (if Claim Name Blinding is to be used for this claim).
+ * Create an array of three elements in this order:
+   1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number.
+   2. The claim name, or key, as it would be used in a regular JWT. This MUST be a string.
+   3. The claim's value, as it would be used in a regular JWT. The value MAY be of any type that is allowed in JSON, including numbers, strings, booleans, arrays, and objects.
+ * JSON-encode the array such that an UTF-8 string is produced.
+ * base64url-encode the byte representation of the UTF-8 string, producing a US-ASCII [@RFC0020] string. This string is the Disclosure.
 
-The following is an example for a JSON literal without Claim Name Blinding:
+The following example illustrates the steps described above.
 
-```
-{"s": "6qMQvRL5haj", "v": "Peter"}
+The array is created as follows:
+```json
+["6qMQvRL5haj", "family_name", "Möbius"]
 ```
 
-The following is an example for a JSON literal with Claim Name Blinding:
+The resulting Disclosure would be `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0`.
 
-```
-{"s": "6qMQvRL5haj", "v": "Peter", "n": "given_name"}
-```
+Note that the JSON encoding of the object is not canonicalized, so variations in white space, encoding
+of Unicode characters, and ordering of object properties are allowed. For example, the following strings
+are all valid and encode the same claim value:
+
+ * A different way to encode the umlaut: `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNXHUwMGY2Yml1cyJd`
```

Comment:
(I did not know the term umlaut...)

Reply:
We can replace it with "letter with crazy german-looking dots on top"? ;-)

Reply:
```suggestion
 * A different way to encode the umlaut (two dots `¨` placed over the letter): `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNXHUwMGY2Yml1cyJd`
```